backends-velox/src/main/scala/org/apache/gluten/extension/HLLRewriteRule.scala

```diff
@@ -25,37 +25,34 @@ import org.apache.spark.sql.catalyst.expressions.Literal
 import org.apache.spark.sql.catalyst.expressions.aggregate.{AggregateExpression, HyperLogLogPlusPlus}
 import org.apache.spark.sql.catalyst.plans.logical.{Aggregate, LogicalPlan}
 import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.catalyst.trees.TreePattern.AGGREGATE_EXPRESSION
 import org.apache.spark.sql.types._
 
 case class HLLRewriteRule(spark: SparkSession) extends Rule[LogicalPlan] {
   override def apply(plan: LogicalPlan): LogicalPlan = LogicalPlanSelector.maybe(spark, plan) {
-    plan.resolveOperatorsUp {
+    plan.transformUp {
```

Comment:
Could you clarify the reason for replacing resolveOperatorsUp with transformUp?

Reply:
In Spark, the `resolveOperatorsUp` used to analysis rule, but `transformUp` used to optimize rule.

Reply:
In more detail, as I understand, use `resolveOperatorsUp` here may cause errors. `resolveOperatorsUp` is similar to `transformUp`, but it will not process nodes which _analyzed is true. @PHILO-HE 
In addition, the same problem exists in `CHAggregateFunctionRewriteRule`. @beliefer

Reply:
@PHILO-HE @zml1206 sorry, I don't quite understand the background information about replacing `resolveOperatorsUp` with `transformUp`. Do we have any issue in velox which could be reproduced ?

Reply:
@beliefer How about `plan.transformUpWithPruning(_.containsPattern(AGGREGATE))` ?

Reply:
@zml1206 Thank you.